src/java/org/apache/cassandra/service/accord/AccordJournal.java

```diff
@@ -1271,9 +1259,9 @@ private static abstract class AuxiliaryRecord
 
     private static final class FrameRecord extends AuxiliaryRecord
     {
-        final Pointer[] pointers;
+        final List<Pointer> pointers;
 
-        FrameRecord(Timestamp timestamp, Pointer[] pointers)
+        FrameRecord(Timestamp timestamp, List<Pointer> pointers)
```

Comment:
It looks like a frame record is basically just a list of pointers, and the timestamp in the frame record is unused right now?

For the pointers could you use collection serializer instead of hand rolling the serialization?

Would it make sense to only have the list of pointers and no frame record class?

Reply:
We do need the key later. Did switch to the collection serializer, cheers.